Step 5: Google Cloud Datastore

zghera opened the pull request:
Extension to comments system built in #13. Now, instead of storing the comments in a List, the comments are stored and loaded from the Google Cloud Datastore.

The original java servlet was split into two servlets: one responsible for post requests (NewComment) and one for get requests (ListComments). NewComment enters comments that were submitted via the html form into the datastore as a new entity of the Comment kind. Once re-directed to the page, the ListComment servlet is called via a get request. A query is made to the datastore for all of the entities in the Comment kind. These comments are converted to a list contained in a JSON string and written to a page that the javascript fetches from.

zghera committed:
Add datastore dependency to pom.xml

zghera committed:
Add functionality to push new comments to datastore.

zghera committed:
Add functionality to load comments from Datastore.

The one java servlet was split into two servlets: one responsible for post requests (NewComment) and one for get requests (ListComments). NewComment enters comments that were submitted via the html form into the datastore as a new entity of the Comment kind. Once re-directed to the page, the ListComment servlet is called via a get request. A query is made to the datastore for all of the entities in the Comment kind. These comments are converted to a list contained in a JSON string and written to a page that the javascript fetches from.

zghera committed:
Integrate formatting changes from commit 8c8b4b8.

zghera committed:
JSDoc style function comments.

zghera committed:
Add (more) proper Doc Comments.

zghera committed:
Remove unused line in method comments.

zghera committed:
Period usage consistancy.

beiebi0 commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:
```diff
@@ -0,0 +1,74 @@
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Is this the sample code they provide? Moving datastore to be a static variable to the servlet makes more sense to me.

zghera replied:
Yes (other than the kind that is queried for was changes to Comment). However I think using the datastore is necessary since the whole purpose of step 5 is to move away from a static class variable to using a database. So is this okay or did I misunderstand your second sentence?

beiebi0 replied:
yes I found the sample code. I mean datastore the service itself can be moved out of the doGet function here. never mind. lgtm now. I need to do a little more research for supporting my argument,

beiebi0 commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java:
```diff
@@ -0,0 +1,74 @@
+    List<String> comments = new ArrayList<>();  
+    for (Entity commentEntity : results.asIterable()) {
+      String comment = (String) commentEntity.getProperty("text");
+
```
nit: empty line seems not necessary here

zghera replied:
Done

zghera committed:
Remove unnecessary empty line.

zghera committed:
Instantiate datastore a static class variable rather than in the doGet/Post method.

Moving the datastore instance to be a class variable ensures that it is only created once for the lifetime of use rather than creating a new datastore instance on each GET and POST request.

zghera committed:
Fix catch placement and indentation on server-dev.js.

zghera committed:
JavaDoc styling.

zghera committed:
Added issue/todo to do input verification for security in the future.

zghera committed:
Merge branch 'master' into step-5-datastore
